consensus/istanbul/backend/api.go

```diff
@@ -248,17 +281,13 @@ func (api *APIExtension) getConsensusInfo(block *types.Block) (ConsensusInfo, er
 
 	// get origin proposer at 0 round.
 	originProposer := common.Address{}
-	lastProposer := common.Address{}
-	// TODO-Klaytn add the logic to get the last proposer for other policies. Weighted Random doesn't need it.
-	if istanbul.ProposerPolicy(snap.Policy) == istanbul.WeightedRandom {
-		newValSet := snap.ValSet.Copy()
-		newValSet.CalcProposer(lastProposer, 0)
-		originProposer = newValSet.GetProposer().Address()
-	} else {
-		logger.Warn("getConsensusInfo does not support to get origin proposal on", "proposerPolicy", snap.Policy)
-	}
+	lastProposer := api.istanbul.GetProposer(blockNumber - 1)
+
+	newValSet := snap.ValSet.Copy()
+	newValSet.CalcProposer(lastProposer, 0)
+	originProposer = newValSet.GetProposer().Address()
```

Comment:
Why don't you use the snapshot of the target block to calculate originProposer like `targetSnap.CalcProposer(lasstProposer, 0)`
And, we need a comment describing `originProposer` that may confuse other developers. Or, we can rename it to `round0Proposer` or `r0Proposer`.

Reply:
The reason why this uses the parent snapshot is same with above comments.

For `originProposer`,  I think it is good suggestion, I will make another PR to change the variable name and RPC output.